src/common/Avatar.js

```diff
@@ -22,7 +22,7 @@ const componentStyles = StyleSheet.create({
 });
 
 type Props = {
-  avatarUrl: string,
+  avatarUrl: string | null,
```

Comment:
Use a [maybe type](https://flow.org/en/docs/types/maybe/) here like `?string`.

Reply:
I considered that but a maybe type accepts the values `string`, `null` and `undefined`. In our codebase, only `string` and `null` seem to be used for avatarUrl, and I wanted to be as strict about that as possible.

Reply:
We definitely have the concept of `undefined` when talking about avatars. I am not so sure about `null` actually. 
Since `null` and `undefined` are often used interchangeably and officially there is no good separation in their meaning (though usually `undefined` means 'missing' or not yet defined, while `null` is a bit more explicit, but more rarely used) using the `?` as we do everywhere else would make sense.

Reply:
Okay, will use `?` then.

But are you sure avatars can actually be undefined?

Here is an example:

![image](https://user-images.githubusercontent.com/7950151/43153144-beda65f8-8f70-11e8-9984-7d4f8594daf0.png)

I wonder in what place of the code this could change to undefined.

Reply:
I wonder the same. I am not sure if this used to not be included at all, but my preference would be to not include any key that has no data (even better, any key that has the default values)

Reply:
It's unfortunate that the one terse syntax Flow gives us here includes both null and undefined, when it's rare that we really want to allow both.

I'm happy to experiment with writing out things like `string | null` when in fact that's what we mean. I wouldn't want to go and do a massive sweep over the codebase for it right now, but we'll see how it looks after playing with it in a few places.

Because I don't see anywhere that we set this to null, I'm assuming that that's the way the server represents it for a user that hasn't set an avatar. That's pretty natural coming from Python -- the row in the database has `NULL` there, so it becomes `None` in Python, and JSON encoders translate that to `null`.

Reply:
Aha, in fact -- this is right there in [the API doc](https://zulipchat.com/api/get-all-users#response) that I linked to in the jsdoc for that type :grin:

> avatar_url: URL to the user's gravatar. `None` if the client_gravatar query paramater was set to  True.

That isn't quite right in a couple of ways -- it's `null`, not None, because this is JSON and not Python, and it's still non-null when the user has set a non-Gravatar avatar. But there it is -- it can be null. (And we do pass `client_gravatar: true`.)